src/main/java/com/google/api/core/ApiFutures.java

```diff
@@ -99,6 +100,21 @@ public void onSuccess(V v) {
                 })));
   }
 
+  public static <I, O> ApiFuture<O> transformAsync(
+      ApiFuture<I> input, final ApiAsyncFunction<I, O> function) {
```

Comment:
This is slightly different from guava, which accepts `AsyncFunction<? super I, ? extends O>`. I have not been able to support this signature without unsafe type casting, so I think we better make the API a little less powerful to avoid the risk of mysterious `ClassCastException`s at runtime. If we later figure out a way to do this, it's an additive change anyway.